crates/oxc_linter/src/rules/eslint/no_restricted_imports.rs

```diff
@@ -1758,50 +1866,50 @@ fn test() {
                 }]
             }])),
         ),
-        // (
-        //     r#"import * as AllowedObject from "foo/bar";"#,
-        //     Some(serde_json::json!([{
-        //         "patterns": [{
-        //             "group": ["foo/*"],
-        //             "allowImportNamePattern": "^Allow"
-        //         }]
-        //     }])),
-        // ),
-        // (
-        //     r#"import * as AllowedObject from "foo/bar";"#,
-        //     Some(serde_json::json!([{
-        //         "patterns": [{
-        //             "group": ["foo/*"],
-        //             "allowImportNamePattern": "^Allow",
-        //             "message": r#"Only import names starting with "Allow" are allowed to be imported from "foo"."#
-        //         }]
-        //     }])),
-        // ),
+        (
+            r#"import * as AllowedObject from "foo/bar";"#,
+            Some(serde_json::json!([{
+                "patterns": [{
+                    "group": ["foo/*"],
+                    "allowImportNamePattern": "^Allow"
+                }]
+            }])),
+        ),
+        (
+            r#"import * as AllowedObject from "foo/bar";"#,
+            Some(serde_json::json!([{
+                "patterns": [{
+                    "group": ["foo/*"],
+                    "allowImportNamePattern": "^Allow",
+                    "message": r#"Only import names starting with "Allow" are allowed to be imported from "foo"."#
+                }]
+            }])),
+        ),
         // (
         //     r#"import withPatterns from "foo/baz";"#,
         //     Some(
         //         serde_json::json!([{ "patterns": [{ "regex": "foo/(?!bar)", "message": "foo is forbidden, use bar instead" }] }]),
         //     ),
```

Comment:
Can we support this test now or does this still not work?

Reply:
No this is a negative lookahead and not supported by rust regex